lib/src/sprite_animation.dart

```diff
@@ -78,7 +78,7 @@ class SpriteAnimationData {
           texturePosition: texturePosition,
           textureSize: textureSize,
           loop: loop,
-          stepTimes: stepTime != null ? List.filled(amount, stepTime) : null,
+          stepTimes: List.filled(amount, stepTime),
```

Comment:
stepTime is @required

Reply:
also there is an assert not null on the other constructor

Reply:
if you leave like this, the assert will not work because the list will be filled with null values, and a List of null is not a null list.

I explicit made this ternary to make the code safer.

This should be reverted. And probably have a test added to check that.

Reply:
but then if we want that we should add an assertion on this constructor making sure stepTime is not null. it's quite weird to explicitly add a null check if null is forbidden.

Reply:
If we do that assert, checking if all the positions on the list have a value, that indeed would be better. Its up to you, or revert the change, or add that new assertion, I am fine with both

Reply:
you don't need to check every position because they are all the same (stepTime). let me add the assert and you will see

Reply:
had to convert to a factory because of a silly `assert_in_redirecting_constructor` restriction of dart. no idea why that is the case but factory works just as well.